src/stack.cairo

```diff
@@ -47,6 +50,10 @@ pub impl ScriptStackImpl of ScriptStackTrait {
 
     fn pop_int(ref self: ScriptStack) -> Result<i64, felt252> {
         let value = self.pop_byte_array()?;
+        if ScriptNum::unwrap(value.clone()) > MAX_INT32.into()
```

Comment:
Hey sorry, could you do this check inside of `ScriptNum::unwrap` so we don't need to unwrap multiple times ( it is pretty expensive ). Also it would make our `unwrap` more close to `MakeScriptNum` from btcd